### ISSUE TITLE
mAP = 0 on custom images

### ISSUE THREAD

**Issue by siatpat** [AlexeyAB/darknet#4902]
Hello @AlexeyAB, I am facing the following issue that for my custom dataset the avg loss is going down but the mAP is still 0.
I have ~ 12000 train images from with 7000 have defects I want to identify. I have 4 classes of defects I am training for, I have around 5000 images in the test set.

I am training on EC2 with 4 GPUs. I have first trained for 1000 iterations as you suggested on one GPU and now I am training for 8000 iterations on all 4 GPUs. The avg loss is going down but mAP is still 0, and I don't know what I should check. 

![darknet-mAP](https://user-images.githubusercontent.com/5452002/75096095-679fb380-559c-11ea-8b20-b821b7be814c.PNG)

All my images are 1600 x 256 and I have kept them that way.
I have modified anchors to used calculated anchors based on my datasets
anchors = 20, 254, 31, 254, 70, 253, 27,103, 25, 39, 16, 18, 56, 56, 155, 89, 434, 182

In cfg file I have
Batch = 64 
subdivisions = 16 
and appart from doing the suggested modifications when training on own dataset, I have only played with max_batches.

I have check with -show_imgs that bounding boxes were showing properly

Do you have any suggestions on why the mAP is showing 0.

Regards

**Comment by AlexeyAB** [comment]
> I have modified anchors to used calculated anchors based on my datasets
> anchors = 20, 254, 31, 254, 70, 253, 27,103, 25, 39, 16, 18, 56, 56, 155, 89, 434, 182

Don't change anchors.
Check your mAP on Training dataset.

**Comment by hbiserinska** [comment]
Hello @AlexeyAB 

Any idea why mAP drops to 0% out of the blue.
I am using yolov4
Training with `! ./darknet detector train data/obj.data cfg/yolov4-custom.cfg backup/yolov4-custom_last.weights -dont_show -map`
![madian](https://user-images.githubusercontent.com/61693348/81385659-694d3200-911c-11ea-9346-14a2d4d24e87.png)

**Comment by AlexeyAB** [comment]
* Attach cfg-file in zip archive.
* Check your dataset, may be some annotations are wrong.
* Show content of files bad.list and bad_label.list near with the ./darknet executable file
